### DISCUSSION TITLE
Dates and date ranges

### DISCUSSION THREAD

**Discussion by hawkeye64** [urql-graphql/urql#3074] [Q&A]
My apologies for having to ask some questions. I am fairly new to urql and GraphQL in general.

I want the user to be able to set a start date or an end date. Either can be used and both are options.
Start date is any dates less than (or equal) than that date
End date is any dates greater (or equal) than that date
If both are used, than is a date range (between those dates)

Here is what I have already and it works (not the dates though). I have tried several things and the returned rows with date sometimes fall outside the date range.

```
query MyQuery(
  $first: Int! = 10,
  $after: String,
  $id_eq: String,
  $assetTo_eq: String,
  $assetFrom_eq: String,
  $timestamp_gte: DateTime,
  $timestamp_lte: DateTime) {
    swapsConnection(
      orderBy: blockNumber_DESC,
      first: $first,
      after: $after,
      where: {
        type_not_eq: "Limit",
        account: {id_eq: $id_eq},
        timestamp_lte: $timestamp_lte,
        timestamp_gte: $timestamp_gte,
        AND: {assetFrom_eq: $assetFrom_eq, OR: {assetTo_eq: $assetTo_eq}}}) {
```
As you can see, I am already using an `AND` clause, which seems to be working well. I just don't know how to fit in the DateTime pieces properly.
Any help would be greatly appreciated. Even some sort of documentation that explains how to deal with date ranges would be fine.

**Comment by kitten** [comment]
I'm sorry but I believe this question isn't meant for urql? This has entirely to do with your GraphQL API, schema, or how you deal with times.

You also haven't really said what you're struggling with.

I'm ok to chat about this on our Discord if you have a specific question.
However it seems there's nothing really here that I could help with and it doesn't really seem to be a question that would relate to urql 😅

**Reply by hawkeye64** [reply]
Sorry, I didn't even know that. I will try to find the Discord link. Thanks.
